src/pip/_internal/commands/show.py

```diff
@@ -122,47 +146,104 @@ def search_packages_info(query):
+
+        information.append(OrderedDict(package_info))
+
+    for package in information:
+        for key, value in package.items():
+            if key == 'Classifiers':
+                logger.info("%s:", key)
+                for classifier in value:
+                    logger.info("  %s", classifier)
+            elif key == 'Entry-points':
+                logger.info("%s:", key)
+                for entry_point, entry_point_info in value.items():
+                    logger.info("  [%s]", entry_point)
+                    for x, y in entry_point_info.items():
+                        logger.info("  %s = %s", x, y)
+            elif isinstance(value, list):
+                logger.info("%s: %s", key, ", ".join(value))
+            else:
+                logger.info("%s: %s", key, value)
+        if information.index(package) < (len(information) - 1):
             logger.info("---")
 
-        name = dist.get('name', '')
-        required_by = [
-            pkg.project_name for pkg in pkg_resources.working_set
-            if name in [required.name for required in pkg.requires()]
-        ]
-
-        logger.info("Name: %s", name)
-        logger.info("Version: %s", dist.get('version', ''))
-        logger.info("Summary: %s", dist.get('summary', ''))
-        logger.info("Home-page: %s", dist.get('home-page', ''))
-        logger.info("Author: %s", dist.get('author', ''))
-        logger.info("Author-email: %s", dist.get('author-email', ''))
-        logger.info("License: %s", dist.get('license', ''))
-        logger.info("Location: %s", dist.get('location', ''))
-        logger.info("Requires: %s", ', '.join(dist.get('requires', [])))
-        logger.info("Required-by: %s", ', '.join(required_by))
-
-        if verbose:
-            logger.info("Metadata-Version: %s",
-                        dist.get('metadata-version', ''))
-            logger.info("Installer: %s", dist.get('installer', ''))
-            logger.info("Classifiers:")
-            for classifier in dist.get('classifiers', []):
-                logger.info("  %s", classifier)
-            logger.info("Entry-points:")
-            for entry in dist.get('entry_points', []):
-                logger.info("  %s", entry.strip())
-        if list_files:
-            logger.info("Files:")
-            for line in dist.get('files', []):
-                logger.info("  %s", line.strip())
-            if "files" not in dist:
-                logger.info("Cannot locate installed-files.txt")
+    return results_printed
+
+
+def print_json(distributions, list_files=False, verbose=False):
+    """
+    Print in JSON format the information from installed distributions found.
+    """
+    information = []
+    results_printed = False
+
+    for dist in distributions:
+        results_printed = True
+
+        package_info = get_package_info(dist, list_files, verbose)
+
+        information.append(OrderedDict(package_info))
```

Comment:
The above four lines appear identically in both `print_json()` and `print_header_format()`. I believe there should be a single `print_distributions()` function called from `run()` containing this logic and that accepts an additional argument controlling the output format.

Reply:
Done! :)

Reply:
Okay, but I had suggested a single print function called from `run()` containing this logic and that accepts an argument controlling the output format. This approach will make testing easier because you won't need to go through `Command.run()` to test it.

Reply:
Hi, I've created the `print_distributions()` function that is called from `run()`.  :)

Sorry for the delay! And I've lost a bit the context of what we were hoping to achieve with the tests -- would you mind explaining?